compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -701,32 +674,7 @@ impl<'tcx> Validator<'_, 'tcx> {
                 }
```

Comment:
`place` gets mutated here, so your refactor has changed the logic in ways that's difficult to predict.

Reply:
Place gets rebound here, yeah... but on the second time reading this I agree I don't understand this code well enough. It removes a top-level `*` but only if the thing right below it is a reference... but that changes its type?!?

I'll move the `validate_ref` change to a separate PR once https://github.com/rust-lang/rust/pull/75585 lands.